File: 0257-binary-tree-paths/0257-binary-tree-paths.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void recursion(TreeNode* root,string ds,vector<string>& ans){
        if(root == NULL) return;
        if(root->left == NULL && root->right == NULL){
            ds+=to_string(root->val);
            ans.push_back(ds);
            return;
        }
        ds+=to_string(root->val);
        ds+='-';
        ds+='>';
        recursion(root->left,ds,ans);
        recursion(root->right,ds,ans);
        return;
    }
    
    vector<string> binaryTreePaths(TreeNode* root) {
        vector<string> ans;
        string ds ="";
        recursion(root,ds,ans);
        return ans;
        
    }
};

File: 0257-binary-tree-paths/0257-binary-tree-paths.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def binaryTreePaths(self, root: Optional[TreeNode]) -> List[str]:
        def dfs(root, path, res):
            if not root.left and not root.right:
                res.append(('->'.join([str(char) for char in path]) + '->' if path else "" ) + str(root.val))
                return
            if root.left:
                dfs(root.left, path + [root.val], res)
            if root.right:
                dfs(root.right, path + [root.val], res)
        
        res = []
        dfs(root, [], res)
        return res